src/services/levels.ts

```diff
@@ -0,0 +1,27 @@
+import { SeriesVisibilityChangeMode } from '@grafana/ui';
+
+export const ALL_LEVELS = ['logs', 'debug', 'info', 'warn', 'error', 'crit'];
```

Comment:
Does this work with numeric log lines as well?

Reply:
That's a good question. I worked on the assumption that these are the levels we will get from `detected_level`, but it wouldn't hurt to double check.

Reply:
Well, there's no defined list. Every value of a `level` label can be the value of `detected_level`.

Reply:
I was just writing. Based on [this code](https://github.com/grafana/loki/blob/dbc0e2489cb7d88f04155f4087e6e40a7dc92ff3/pkg/distributor/distributor.go#L402-L416), it would grab the level 1) from a label 2) from metadata 3) detect it (returning normalized names). 

I have an idea, but otherwise we can't make assumptions about possible levels, so we need to adjust.

Reply:
Updated. We now read all the levels from the series, so it should support whatever Loki sends as `detected_level`.